src/search.py

```diff
@@ -0,0 +1,62 @@
+from collections import deque
+import st
+
+def LCP(a: str, b: str) -> int:
+    """
+    Finds the longest common prefix of a and b
+    
+    returns: 
+    If there is no prefix returns 0 otherwise returns length of prefix
+    """
+    i = 0
+    while(i < len(a) and i < len(b)):
+        if a[i] != b[i]:
+            return i
+        i+=1
+    return i
```

Comment:
This is exactly what we want. You can simplify it a little by iterating over the input instead of indices into it. If you `zip()` the input, you get an iterator though the strings (or really anything you provide that can iterate), and you can run through it until you find a difference or reach the end. If you use `enumerate()`, you get the index as well.

```python
def LCP(x: Iterator[Any], y: Iterator[Any]) -> int:
  i = 0
  for i, (a, b) in enumerate(zip(x, y)):
    if a != b: break
  return i
```

With this function, it always annoys the shit out of me that you have to set `i` before the `for` loop, but it is necessary. If one of the iterators is empty, we don't enter the loop, and then `i` won't get set. So we do need to initialise it.